### PR TITLE
Fix for NodeJS with ESM modules

### PR DESCRIPTION
**Pull request by ggetz**
Fixes https://github.com/CesiumGS/cesium/issues/10684

NodeJS was failing to run ESM modules mainly because of a recent update to `zip.js` which requires `TransformStream` to be defined. Looks like [the zip.js maintainer addressed this and suggests](https://github.com/gildas-lormeau/zip.js/issues/340#issuecomment-1177329994) either locking to `2.4.x` or adding a polyfill for `TransformStream`. Here, I've locked the package version.

To prevent related issues from going unnoticed in the future, I added a `.mjs` version of `test.cjs`, which is now run as part of CI.

In doing so, I also caught two other issues. One was a faulty import, and the other was the use of `require` in `Resource.js`. If I understand correctly, this is code that should only ever be executed in a NodeJS environment, and can be replaced with a dynamic import , satisfying both the ESM and CJS use cases.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @ggetz!
* :heavy_check_mark: Signed CLA found.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

File: .travis.yml [hunk 1]
```diff
@@ -30,5 +30,6 @@ script:
   - node -e "const Cesium = require('./');"
   - NODE_ENV=development node Specs/test.cjs
   - NODE_ENV=production node Specs/test.cjs
+  - node Specs/test.mjs
```
Comment (sanjeetsuhag) [inline .travis.yml:33]: Do we need to test this in both `production` and `development` mode too, like we do with `test.cjs`? Looking at `test.cjs`, it is not obvious to me what the point of setting the environment to `production` is.

File: Source/Core/Resource.js [hunk 2]
```diff
@@ -2059,59 +2059,65 @@ function loadWithHttpRequest(
...
-    .on("response", function (res) {
-      if (res.statusCode < 200 || res.statusCode >= 300) {
-        deferred.reject(
-          new RequestErrorEvent(res.statusCode, res, res.headers)
-        );
-        return;
-      }
+  let URL;
+  let zlib;
+  Promise.all([import("url"), import("zlib")])
```
Comment (sanjeetsuhag) [inline Source/Core/Resource.js:2064]: Just to make sure - these imports are cached, right?

**Review by sanjeetsuhag — COMMENTED** [review]
This looks good to me, @ggetz. I just have a couple of things I wanted to clarify.

File: .travis.yml (hunk 1 above)
Comment (ggetz) [reply .travis.yml:33]: Good question! So `test.cjs` imports from the root `cesium` module. In out `package.json`, we've defined [conditional exports](https://nodejs.org/api/packages.html#conditional-exports) such that requiring or importing `cesium` which grab a different file depending on the environment. In CJS with a require, we redirect to `index.cjs`, which in turn requires either the minified release or unminified release depending on whether or not the environment is production. (We've recently had [some discussion whether or not `index.cjs` and the runtime operations are needed or not](https://github.com/CesiumGS/cesium/issues/9212#issuecomment-1218449631).)

MJS on the other hand can handle ESM directly, so imports from `Source/Cesium.js` file. We could (and probably should) have `test.mjs` import from the root `cesium`, and it should be functionally equivalent to `Source/Cesium.js`, and so would also verifies our `package.json` configuration as well.

File: Source/Core/Resource.js (hunk 2 above)
Comment (ggetz) [reply Source/Core/Resource.js:2064]: This code should only ever be executed in environments where `XMLHttpRequest` is undefined, which is not in the browser but in NodeJS. In that case, the are dynamically imported from the `node_modules` directory.

**Review by sanjeetsuhag — APPROVED** [review]
Looks good. Thanks @ggetz!

**Pull request merged**